lib/engine/game/g_18_india/game.rb

```diff
@@ -621,6 +628,167 @@ def legal_tile_rotation?(_entity, _hex, _tile)
+          return 0 if non_variable_stops.empty?
+
+          max_non_variable_value = non_variable_stops.map { |e| e.revenue_to_render - 20 }.max
+          stop_location_names = stops.map { |stop| stop.tile.location_name }.compact
+          variable_city_stops = stop_location_names & VARIABLE_CITY_NAMES
+
+          train_multiplier = route.train.multiplier || 1
+          LOGGER.debug "variable_city_revenue >> train_multiplier: #{train_multiplier}  "\
+                       "variable_city_stops: #{variable_city_stops}  max_non_variable_value: #{max_non_variable_value}"
+
+          variable_city_stops.count * [max_non_variable_value, 0].max * train_multiplier
+        end
+
+        def connection_bonus(route, _stops)
+          visited_location_names = route.visited_stops.map { |stop| stop.tile.location_name }.compact
+          return 0 if visited_location_names.count < 2
+
+          LOGGER.debug "connection_bonus >> visited_location_names: #{visited_location_names}"
+          # Delhi, Kochi => 100 [G8, G36]
+          # Karachi, Chennai => 80 [A16, K30]
+          # Lahore, Kolkata => 80 [D3, P17]
+          # Nepal, Mumbai => 70 [M10, D23]
+          revenue = 0
+          revenue += 100 if visited_location_names.include?('DELHI') && visited_location_names.include?('KOCHI')
+          revenue += 80 if visited_location_names.include?('KARACHI') && visited_location_names.include?('CHENNAI')
+          revenue += 80 if visited_location_names.include?('LAHORE') && visited_location_names.include?('KOLKATA')
+          revenue += 70 if visited_location_names.include?('NEPAL') && visited_location_names.include?('MUMBAI')
+          revenue
+        end
+
+        def available_commodities(corporation)
+          @unclaimed_commodities + corporation.commodities
+        end
+
+        # TODO: Add visual indicaiton on VIEW for Corporation card for Claimed Commodities
+        def claim_concession(commodities, corporation)
+          return if corporation.commodities.include?(commodities.first)
+
+          commodities.each do |commodity|
+            @unclaimed_commodities.delete(commodity)
+            corporation.commodities << commodity
+          end
+          @log << "#{corporation.name} claims the #{commodities} concession"
+        end
+
+        def commodity_bonus(route, _stops)
+          visited_names = route.all_hexes.map(&:location_name).compact
+          corporation = route.train.owner
+          commodity_sources = visited_names & available_commodities(corporation)
+          return 0 unless commodity_sources.count.positive?
+
+          revenue = 0
+          commodity_sources.each do |source|
+            bonus = 0
+            case source
+            when 'OIL'
+              # OIL => MUMBAI [D23] + 30
+              bonus = 30 if visited_names.include?('MUMBAI')
+              claim_connession(['OIL'], corporation) if bonus.positive?
```

Comment:
Oh whoops, I should have mentioned that this line and the other calls also need the spelling fix for `claim_concession`

Reply:
Fixed the calls to the misnamed methods.